scapy/dadict.py

```diff
@@ -11,13 +11,14 @@
 from __future__ import print_function
 from scapy.error import Scapy_Exception
 import scapy.modules.six as six
+from scapy.compat import *
 
 ###############################
 ## Direct Access dictionary  ##
 ###############################
 
 def fixname(x):
-    if x and x[0] in "0123456789":
+    if x and str(x[0]) in "0123456789":
```

Comment:
If you need str(x[0]) here, you'll probably need str(x) the line after that I suppose.
Also, shouldn't we use `raw()` here instead of `str()`?